cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Query.hs

```diff
@@ -78,101 +80,170 @@ hprop_cli_queries = integrationWorkspace "cli-queries" $ \tempAbsBasePath' -> H.
-            . T.strip
-            . fromString )
-          . drop 2 -- drop header
-          . lines
-          $ stakeDistrOut
-    H.assertWith stakeAddresses $ \sa ->
-      -- Because, by default, 3 stake pools are created
-      length sa == 3
-    -- Light test of the query's answer, the ids should exist:
-    forM_ stakeAddresses $ \(stakePoolId, _) -> do
-      execCli' execConfig [ eraName, "query", "pool-state"
-                          , "--stake-pool-id", T.unpack stakePoolId ]
-    -- to a file
-    let stakePoolsOutFile = work </> "stake-distribution-out.json"
-    H.noteM_ $ execCli' execConfig [ eraName, "query", "stake-distribution"
-                                   , "--out-file", stakePoolsOutFile]
-
-  -- gov-state
-  do
-    -- to stdout
-    execCli' execConfig [ eraName, "query", "gov-state" ]
-      >>=
-        (`H.diffVsGoldenFile`
-            "test/cardano-testnet-test/files/golden/queries/govStateOut.json")
-    -- to a file
-    let govStateOutFile = work </> "gov-state-out.json"
-    H.noteM_ $ execCli' execConfig [ eraName, "query", "gov-state", "--out-file", govStateOutFile]
-    H.diffFileVsGoldenFile
-      govStateOutFile
-      "test/cardano-testnet-test/files/golden/queries/govStateOut.json"
-
-  -- drep-state
-  do
-    -- to stdout
-    -- TODO: deserialize to a Haskell value when
-    -- https://github.com/IntersectMBO/cardano-cli/issues/606 is tackled
-    dreps :: Aeson.Value <- H.noteShowM $ execCliStdoutToJson execConfig [ eraName, "query", "drep-state", "--all-dreps"]
-    assertArrayOfSize dreps 3
-    -- to a file
-    let drepStateOutFile = work </> "drep-state-out.json"
-    H.noteM_ $ execCli' execConfig [ eraName, "query", "drep-state", "--all-dreps"
-                                   , "--out-file", drepStateOutFile]
-    _ :: Aeson.Value <- H.readJsonFileOk drepStateOutFile
-    pure ()
-
-  H.success
+  forallQueryCommands (\case
+
+    TestQueryLeadershipScheduleCmd -> do
+      -- leadership-schedule
+      pure ()
+
+    TestQueryProtocolParametersCmd ->
+      -- protocol-parameters
+      do
+        -- to stdout
+        protocolParametersOut <- execCli' execConfig [ eraName, "query", "protocol-parameters" ]
+        H.diffVsGoldenFile
+          protocolParametersOut
+          "test/cardano-testnet-test/files/golden/queries/protocolParametersOut.txt"
+        -- protocol-parameters to a file
+        let protocolParametersOutFile = work </> "protocol-parameters-out.json"
+        H.noteM_ $ execCli' execConfig [ eraName, "query", "protocol-parameters"
+                                       , "--out-file", protocolParametersOutFile ]
+        H.diffFileVsGoldenFile
+          protocolParametersOutFile
+          "test/cardano-testnet-test/files/golden/queries/protocolParametersFileOut.json"
+
+    TestQueryConstitutionHashCmd -> do
+      -- constitution-hash
+      -- Currently disabled (not accessible from the command line)
```

Comment:
We need to investigate why this is disabled.

Reply:
Yes, @disassembler [suggests](https://input-output-rnd.slack.com/archives/C031GFLC3DF/p1721363023680529?thread_ts=1721353986.551329&cid=C031GFLC3DF) this functionality may already be provided by `query constitution`. If that is the case we may just delete the query

Reply:
I have created [this ticket](https://input-output-rnd.slack.com/lists/T0N639Z4N/F078RTQGB1T?record_id=Rec07GQN6EVQ8)